usr/src/uts/common/fs/zfs/spa.c

```diff
@@ -6985,6 +6987,14 @@ spa_event_create(spa_t *spa, vdev_t *vd, const char *name)
 		}
 	}
 
+	if (hist_nvl) {
```

Comment:
after review from joshc bringing up the same issue, I agree that it should be `if (hist_nvl !=null) { ... }` as it explicitly conveys our intent - will change.